### PR TITLE
feat: replace custom context prop with custom queryClient instance

### PR DESCRIPTION
**Pull request by DamianOsipiuk**
BREAKING CHANGE: custom context has been removed from hooks in favor of passing custom queryClient instance

Closes: #4682

TODO:
- [x] Migration guide entry
- [x] Add replaceable content section in migration guide for framework specific section

### CODE REVIEW

**Review by TkDodo — COMMENTED** [review]
thank you, this looks good.
we'd need an entry in the migration guide please, and maybe it makes sense to keep one test somewhere that tests with a custom queryClient being passed around?

File: packages/solid-query/src/createMutation.ts [hunk 1]
```diff
@@ -17,11 +18,12 @@ export function createMutation<
   TContext = unknown,
 >(
   options: CreateMutationOptions<TData, TError, TVariables, TContext>,
+  queryClient?: () => QueryClient,
 ): CreateMutationResult<TData, TError, TVariables, TContext> {
-  const queryClient = useQueryClient({ context: options().context })
+  const client = useQueryClient(queryClient?.())
```
Comment (DamianOsipiuk) [inline packages/solid-query/src/createMutation.ts:23]: @ardeora In createBaseQuery we are wrapping useQueryClient with createMemo.
However in createMutation and createQueries we do not do this.
Should we add createMemo to those hooks as well?

**Comment by DamianOsipiuk** [comment]
@TkDodo PR should be ready now

**Review by TkDodo — APPROVED** [review]

File: packages/solid-query/src/createMutation.ts (hunk 1 above)
Comment (ardeora) [reply packages/solid-query/src/createMutation.ts:23]: This is fine for now 😄 The memoed queryClient isn't really fully reactive yet.  I'll create a separate PR to wire in the reactivity for the other queryClients too

**Comment by ardeora** [comment]
This looks awesome! I love this change!

**Review by ardeora — APPROVED** [review]

**Pull request merged**